lib/livebook_cli/server.ex

```diff
@@ -141,25 +146,37 @@ defmodule LivebookCLI.Server do
     end
   end
 
-  defp open_from_options(base_url, opts, []) do
-    if opts[:open] do
-      Livebook.Utils.browser_open(base_url)
-    end
+  defp open_from_args(base_url, []) do
+    Livebook.Utils.browser_open(base_url)
   end
 
-  defp open_from_options(base_url, _opts, ["new"]) do
+  defp open_from_args(base_url, ["new"]) do
     base_url
     |> append_path("/explore/notebooks/new")
     |> Livebook.Utils.browser_open()
   end
 
-  defp open_from_options(base_url, _opts, [url]) do
-    base_url
-    |> Livebook.Utils.notebook_import_url(url)
-    |> Livebook.Utils.browser_open()
+  defp open_from_args(base_url, [url_or_file_or_dir]) do
+    cond do
+      File.regular?(url_or_file_or_dir) || File.dir?(url_or_file_or_dir) ->
+        base_url
+        |> update_query(%{"file" => url_or_file_or_dir})
+        |> Livebook.Utils.browser_open()
+
+      %{host: nil} = URI.parse(url_or_file_or_dir) ->
```

Comment:
This code will error as soon as you pass a URL with a host because `=` will try to match them.

My suggestion would to be reorder clauses like this:

```
url = URI.parse(url_or_file_or_dir)

cond do
  url.scheme in ~w(http https file) ->
    ...

  File.dir?(...) ->
    ...
  File.regular?(...) ->
    ...
  true ->
    ...
end
```